cli/tokens.go

```diff
@@ -63,8 +63,26 @@ func (r *RootCmd) createToken() *serpent.Command {
 			if user != "" {
 				userID = user
 			}
+
+			var parsedLifetime time.Duration
+			var err error
+
+			if tokenLifetime == "" {
+				tokenConfig, err := client.GetTokenConfig(inv.Context(), userID)
+				if err != nil {
+					return xerrors.Errorf("get token config: %w", err)
+				}
+
+				parsedLifetime = tokenConfig.MaxTokenLifetime
+			} else {
+				parsedLifetime, err = extendedParseDuration(tokenLifetime)
+				if err != nil {
+					return xerrors.Errorf("parse lifetime: %w", err)
+				}
```

Comment:
suggestion: given that we have the token config available, we can check if the requested lifetime is greater than `MaxTokenLifetime` client-side. This may not be completely necessary though, as the API will disallow creating tokens with lifetime greater than `MaximumTokenDuration`. Dealer's choice!

Reply:
Yeah you're completely right - makes sense I just changed it a bit to handle the config fetched ! 👌